editor/utils/dom.js

```diff
@@ -77,17 +145,101 @@ export function placeCaretAtEdge( container, start = false ) {
 		return;
 	}
 
+	if ( ! container.isContentEditable ) {
+		container.focus();
+		return;
+	}
+
 	// Content editables
 	const range = document.createRange();
 	range.selectNodeContents( container );
-	range.collapse( start );
+	range.collapse( reverse );
 	const sel = window.getSelection();
 	sel.removeAllRanges();
 	sel.addRange( range );
 	container.focus();
 }
 
 /**
+ * Polyfill.
+ * Get a collapsed range for a given point.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Document/caretRangeFromPoint
+ *
+ * @param  {Document} doc The document of the range.
+ * @param  {Float}    x   Horizontal position within the current viewport.
+ * @param  {Float}    y   Vertical position within the current viewport.
+ * @return {?Range}       The best range for the given point.
+ */
+function caretRangeFromPoint( doc, x, y ) {
+	if ( doc.caretRangeFromPoint ) {
+		return doc.caretRangeFromPoint( x, y );
+	}
+
+	if ( ! doc.caretPositionFromPoint ) {
+		return null;
+	}
+
+	const point = doc.caretPositionFromPoint( x, y );
+	const range = doc.createRange();
+
+	range.setStart( point.offsetNode, point.offset );
+	range.collapse( true );
+
+	return range;
+}
+
+/**
+ * Places the caret at the top or bottom of a given element.
+ *
+ * @param {Element} container Focusable element.
+ * @param {Boolean} reverse   True for bottom, false for top.
+ * @param {Boolean} noScroll  Set to true to prevent scrolling.
+ */
+export function placeCaretAtVerticalEdge( container, reverse, noScroll ) {
+	const rect = firstVerticalRect;
+
+	if ( ! rect || ! container.isContentEditable ) {
+		placeCaretAtHorizontalEdge( container, reverse );
+		return;
+	}
+
+	const buffer = rect.height / 2;
+	const editableRect = container.getBoundingClientRect();
+	const x = rect.left + ( rect.width / 2 );
+	const y = reverse ? ( editableRect.bottom - buffer ) : ( editableRect.top + buffer );
+	const selection = window.getSelection();
+
+	// Temporary high z-index above toolbars.
+	container.style.zIndex = '10000';
+
+	const range = caretRangeFromPoint( document, x, y );
+
+	container.style.zIndex = null;
```

Comment:
Hm, bummer. :)

Local testing (on this very page) indicates that `container.style.pointerEvents = 'none';` is just as effective. Should we prefer that one instead? I wonder if that way we minimize our chances of generating visual artefacts.

Reply:
How would that be used? On the toolbar element? That would require us to have the toolbar element, which would be less isolated logic. In that case I'd prefer the z-index approach.

Reply:
Yes, it would be on the toolbar — I was looking at this so closely that I didn't weigh the drawback of having to deal with yet another entity. z-index may be the lesser evil, then.